zookeeper-jute/src/main/java/org/apache/jute/Utils.java

```diff
@@ -30,32 +30,6 @@ private Utils() {
         super();
     }
 
-    /**
-     * equals function that actually compares two buffers.
-     *
-     * @param onearray First buffer
-     * @param twoarray Second buffer
-     * @return true if one and two contain exactly the same content, else false.
-     */
-    public static boolean bufEquals(byte[] onearray, byte[] twoarray) {
-        if (onearray == twoarray) {
-            return true;
-        }
-
-        boolean ret = (onearray.length == twoarray.length);
-
-        if (!ret) {
-            return ret;
-        }
-
-        for (int idx = 0; idx < onearray.length; idx++) {
-            if (onearray[idx] != twoarray[idx]) {
-                return false;
-            }
-        }
-        return true;
-    }
-
     public static int compareBytes(byte[] b1, int off1, int len1, byte[] b2, int off2, int len2) {
```

Comment:
Do you want to try to use Arrays.compare introduced in JDK9, in is another hotspot intrinsic ? 
We should go for a multi release jar
I have done it in this project
https://github.com/diennea/herddb/blob/master/herddb-utils/src/main/java10/herddb/utils/CompareBytesUtils.java

It is just a suggestion, I see you like this kind of stuff, the same for me :-)

Reply:
@eolivelli I don't think the group of folks here is quite large enough to be supporting mult-version releases.  I think it's best just to focus on getting everything up to JDK 1.8 spec first and then continue to increment JDK versions over time.